### PR TITLE
Feature 32 integrate vuetify/vuex

### PR DESCRIPTION
**Pull request by dimitrov93**

### CODE REVIEW

**Review by Alekseeybg — APPROVED** [review]
Tested manually - everything worked correctly. Reviewed the code - did not see any issues

File: src/components/ui/SnackBar.vue
Comment (VladoKat) [inline src/components/ui/SnackBar.vue:1]: Why do we need to create this component? Do we need something more than the out-of-the-box one? I don't see a customization here but maybe I am missing something.

File: src/main.js [hunk 1]
```diff
@@ -1,11 +1,11 @@
...
+import { registerPlugins } from "@/plugins";
+import store from "./store/store.js";
+
 const app = createApp(App);
+registerPlugins(app);
 
-app.use(router);
-app.component("base-card", BaseCard);
-app.component("base-button", BaseButton);
+app.use(store);
```
Comment (VladoKat) [inline src/main.js:10]: should we move this to the registerPlugins(app) method? I see the other freameworks are registered there.

File: src/views/Resources.vue [hunk 2]
```diff
@@ -1,57 +1,37 @@
...
-    async fetchResources() {
-      const fetchedResources = await fetchResources();
-      this.resources = fetchedResources;
-    },
-  },
-  mounted() {
-    this.fetchResources();
+    onMounted(() => {
+      store.dispatch("resources/fetchResources");
+    });
```
Comment (VladoKat) [inline src/views/Resources.vue:30]: We don't handle errors thrown by fetchResources here. I think we should like in the products.

File: src/views/Users.vue [hunk 3]
```diff
@@ -1,28 +1,35 @@
...
+  setup() {
+    const store = useStore();
+    const showSnackbar = inject("showSnackbar");
+
+    onMounted(async () => {
+      try {
+        store.dispatch("users/fetchUsers");
+      } catch (error) {
+        showSnackbar({
+          message: error.message || "Failed to fetch users.",
```
Comment (VladoKat) [inline src/views/Users.vue:23]: If I understand it correctly. we are always going to show the error.message if it's there. I think it's always going to be there. I think we should just show the "Failed to fetch users" message. I may be misunderstanding this, though.

**Review by VladoKat — CHANGES_REQUESTED** [review]
Left some comments/questions. Let me know what you think

File: src/App.vue [hunk 4]
```diff
@@ -1,38 +1,77 @@
...
+      <router-view />
+    </v-main>
+    <SnackBar
+      v-model="snackbar.isActive"
+      :isActive="snackbar.isActive"
+      :message="snackbar.message"
+      :color="snackbar.color"
+      :location="snackbar.location"
+      :timeout="snackbar.timeout"
+      :width="snackbar.width"
```
[the remark this answers is not in the document]
Comment (dimitrov93) [reply src/App.vue:13]: fixed

File: src/main.js (hunk 1 above)
Comment (dimitrov93) [reply src/main.js]: fixed

File: src/views/Resources.vue (hunk 2 above)
Comment (dimitrov93) [reply src/views/Resources.vue]: fixed

File: src/views/Users.vue (hunk 3 above)
Comment (dimitrov93) [reply src/views/Users.vue]: fixed

**Review by VladoKat — APPROVED** [review]
Looks good!

**Pull request merged**